packages/flutter_tools/lib/src/commands/assemble.dart

```diff
@@ -118,10 +126,23 @@ class AssembleCommand extends FlutterCommand {
     );
     usesExtraDartFlagOptions(verboseHelp: verboseHelp, useLegacyNames: useLegacyNames);
     usesDartDefineOption(useLegacyNames: useLegacyNames);
+    argParser.addOption(kDeferredComponents);
     argParser.addOption(
       'resource-pool-size',
       help: 'The maximum number of concurrent tasks the build system will run.',
     );
+    argParser.addFlag('verify-deferred-components',
```

Comment:
Pass this through as a build define instead of creating a special argument for it. The build apk/appbundle et cetera commands should have the well documented arg